Forge.TreeWalker/src/TreeNodeContext.cs

```diff
@@ -0,0 +1,102 @@
+//-----------------------------------------------------------------------
+// <copyright file="TreeNodeContext.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeNodeContext class.
+// </summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Forge.TreeWalker
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// The TreeNodeContext class object is passed to Callbacks.BeforeVisitNode and Callbacks.AfterVisitNode.
+    /// </summary>
+    public class TreeNodeContext
+    {
+        /// <summary>
+        /// The unique identifier for this tree walking session.
+        /// </summary>
+        public Guid SessionId { get; private set; }
+
+        /// <summary>
+        /// The tree node key where the Action resides.
+        /// </summary>
+        public string TreeNodeKey { get; private set; }
+
+        /// <summary>
+        /// The dynamic properties of this Node given by the ForgeTree schema.
+        /// </summary>
+        public dynamic Properties { get; private set; }
+
+        /// <summary>
+        /// The dynamic user-defined context object that is able to be referenced when evaluating schema expressions or performing actions.
+        /// </summary>
+        public object UserContext { get; private set; }
+
+        /// <summary>
+        /// The cancellation token.
+        /// </summary>
+        public CancellationToken Token { get; private set; }
+
+        /// <summary>
+        /// The name of the ForgeTree in the JsonSchema.
+        /// </summary>
+        public string TreeName { get; private set; }
+
+        /// <summary>
+        /// The unique identifier for the root/parent tree walking session.
+        /// </summary>
+        public Guid RootSessionId { get; private set; }
```

Comment:
Let's use the summaries from ITreeWalkerCallbacks.

Reply:
fixed

---
In reply to: [506635372](https://github.com/microsoft/Forge/pull/32#discussion_r506635372) [](ancestors = 506635372)